src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -201,14 +202,42 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
             return validName;
         }
 
-        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(Type type)
+        private static IEnumerable<TagHelperAttributeDescriptor> GetAttributeDescriptors(
+            Type type, 
+            ErrorSink errorSink)
         {
-            var properties = type.GetRuntimeProperties().Where(IsValidProperty);
+            var properties = type.GetRuntimeProperties().Where(IsAccessibleProperty);
             var attributeDescriptors = properties.Select(ToAttributeDescriptor);
 
+            attributeDescriptors = attributeDescriptors.Where(
+                descriptor => ValidTagHelperAttributeDescriptor(descriptor, type, errorSink));
+
             return attributeDescriptors;
         }
 
+        private static bool ValidTagHelperAttributeDescriptor(
```

Comment:
The existing `IsValidProperty` method was misleading to begin with which is why i changed it to `IsAccessibleProperty`. Reason why I didn't group this in with it was because I'd have to recalculate the resulting attribute name to find out if it is valid or not. An alternative if you don't like the extra `where` is to do:

```C#
var attributeDescriptors = new List<TagHelperAttributeDescriptor>();
foreach (var property in properties)
{
    var descriptor = ToAttributeDescriptor(property);
    if (ValidTagHelperAttributeDescriptor(descriptor, type, errorSink))
    {
        attributeDescriptors.Add(descriptor)
    }
}
```